File: go/fixture/go-mini/internal/models/heartbeat.go
package models

import "time"

// Heartbeat is a heartbeat.
//
// See docs/snapshots.md for the snapshot format.
type Heartbeat struct {
	DeviceID   string
	Tenant     string
	Status     string
	Version    string
	Tags       []string
	ReceivedAt time.Time
	// caller must ensure the line is validated
	Raw string
}

// Line returns the line.
func (h Heartbeat) Line() string {
	return h.Raw
}

File: py/fixture/py-mini/internal/models/heartbeat.py
"""Heartbeats as devices send them."""

from dataclasses import dataclass
from datetime import datetime


@dataclass
class Heartbeat:
    """Heartbeat is a heartbeat.

    See docs/snapshots.md for the snapshot format.
    """

    device_id: str
    tenant: str
    status: str
    version: str
    tags: list[str]
    received_at: datetime
    # caller must ensure the line is validated
    raw: str

    def line(self) -> str:
        """Return the line."""
        return self.raw
